src/ServiceManager.php

```diff
@@ -737,6 +737,32 @@ private function createLazyServiceDelegatorFactory()
         return $this->lazyServicesDelegator;
     }
 
+    /**
+     * Merge delegators avoiding multiple same delegators for the same service.
+     * It works with strings and class instances.
+     * It's not possible to de-duple anonymous functions
+     *
+     * @param string[][]|Factory\DelegatorFactoryInterface[][] $config
+     * @return string[][]|Factory\DelegatorFactoryInterface[][]
+     */
+    private function mergeDelegators(array $config)
+    {
+        foreach ($config as $key => $delegators) {
+            if (! isset($this->delegators[$key])) {
+                $this->delegators[$key] = $delegators;
+                continue;
+            }
+
+            foreach ($delegators as $delegator) {
+                if (! in_array($delegator, $this->delegators[$key], true)) {
```

Comment:
This `in_array()` is repeated multiple times: potentially an issue as the array grows.

I don't think it's an issue, since collisions are rare, but worth noting.

Up to you if it needs to be adjusted or not.